Si̇ngleplayer Tutorial Refactoring

alptuncs opened the pull request:
- [x] Eksik olan GameManager unit testleri ve BoardManager'da yeni methodların unit testlerinin eklenmesi
- [x] Verilen özelliklere göre Target oluşturan bir Target Factory implement etme
- [x] GameSessionFactory oluşturma
- [x] Fonksiyonların sadece kendi yapması gereken işi yapacağı şekilde ayrılması
- [x] New ile object yaratılan yerlerde Factory kullanılması
- [x] BoardManager Refactoring
- [x] GameManager Refactoring

alptuncs committed:
Target Factory, BoardManager Factory

alptuncs committed:
Merge from main

alptuncs committed:
GameManager Unit Tests

GameManager test edilebilir hale getirildi, GameManager unit testleri artık çalışıyor

SeferMirza commented on Battleships/Battleships/BoardManager.cs:
```diff
@@ -81,198 +83,124 @@ public void RandomPlaceShip(int count, Target ship)
+        private bool TryPlaceShip(Coordinate coordinate, Target ship)
         {
-            if (ship.Size == 1)
+            if (!CanPlaceShip(coordinate, ship))
```
```csharp
if(!CanPlaceShip(coordinate,ship) || !CheckNeighbors(coordinate, ship))
```
şeklinde yaparak tek ifte kontrol yapabilirsin kod kısalır biraz

alptuncs replied:
Düzelttim

SeferMirza commented on Battleships/Board Tests/GameManagerUnitTests.cs:
```diff
@@ -39,11 +39,52 @@ public void Oyunu_Olusturup_Kullanicidan_Koordinat_Ister()
             game.Initialize();
-            game.Play();
+
+            game.UpdateGame("1,1");
+            game.UpdateGame("1,1");
+            game.UpdateGame("1,1");
```
Bu kadar tekrar doğru gelmedi. Ne için bu?

alptuncs replied:
kullanıcıya 30 hak veriyorum oyun başında tüm hakları bitirmek için sürekli 1,1 koordinatı deneniyor gibi olsun diye böyle yapmıştım

alptuncs committed:
Cell & Direction Class

Cell Struct yerine Cell class eklendi, GameManager'da birden fazla iş yapan fonksiyonlar yaptıkları işe göre bölündü, Enum Direction yerine Direction class eklendi

alptuncs committed:
Target Factory

alptuncs committed:
Board Manager Refactoring

alptuncs committed:
MessageFactory Removed

alptuncs committed:
Oyunu Kazanma

Oyunu kazanma koşulu değişti, gemilerin koordinatları listeye alınıyor o listede olan tüm koordinatlar vurulunca oyun kazanılıyor

alptuncs committed:
GameSessionFactory

alptuncs committed:
Boşluk düzeltme

alptuncs committed:
Add files via upload

alptuncs committed:
Name Change

alptuncs committed:
Update README.md

alptuncs committed:
Update README.md

alptuncs committed:
Update README.md

alptuncs committed:
Update README.md

alptuncs committed:
Update README.md

alptuncs committed:
ReadOnly
